scripts/area_plot.py

```diff
@@ -0,0 +1,613 @@
+import argparse
+import warnings
+# from warnings import Warning
+
+from matplotlib import rcParams
+from matplotlib import cm
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+
+"""
+A set of python scripts to generate an area plot.
+"""
+# Sets up the matplotlib paramters so that we can save to be edited in 
+# illustator if a direct conversion is required. Because it just makes life
+# better
+rcParams['pdf.fonttype'] = 42
+rcParams['ps.fonttype'] = 42
+
+# Sets up the order of colors to be used in joint plots.
+colors_order = ['Reds', 'Blues', 'Greens', 'Purples', "Oranges", 'Greys']
+
+over9 = {'Paired', 'Paired_r', 'Set3', 'Set3_r'}
+over8 = over9 | {'Set1', "Pastel1"}
+
+def extract_label_array(table, tax_col, tax_delim='|'):
+    """
+    Converts delimited taxonomy strings into a working table
+    
+    Parameters
+    ----------
+    table : DataFrame
+        A DataFrame with observation on the rows (biom-style table) with 
+        `tax_col` as one of its columns.
+    tax_col : str
+        The column in `table` containig the taxobnomy information
+    tax_delim: str, optional
+        The delimiter between taxonomic groups
+        
+    Returns
+    -------
+    DataFrame
+        The taxonomic strings parsed into n levels
+    """
+    def f_(x):
```

Comment:
Please use a more descriptive function name. No need to use the trailing underscore either in my opinion :man_shrugging:. I'm guessing you're using the trailing underscore to denote a private function, but that convention is a _leading_ underscore in Python. It's not really needed here regardless, I think.

Reply:
Because I'd normally hit this as a lambda function, I tend to think if `f_`. I'd rather just be able to run it as a straight lambda, but there isn't space. So... `f_`